pulp_rpm/tests/functional/utils.py

```diff
@@ -285,3 +285,20 @@ def get_package_repo_path(package_filename):
     return os.path.join(
         PACKAGES_DIRECTORY, package_filename.lower()[0], package_filename
     )
+
+
+class PulpTestCase(TestCase):
+    """Pulp customized test case."""
+
+    def doCleanups(self):
+        """
+        Execute all cleanup functions and waits the deletion tasks.
+
+        Normally called for you after tearDown.
+        """
+        output = super().doCleanups()
+        running_tasks = tasks.list(state="running", name__contains="delete")
+        while running_tasks.count:
+            sleep(0.3)
+            running_tasks = tasks.list(state="running", name__contains="delete")
+        return output
```

Comment:
As the other tests were using [addCleanup](https://docs.python.org/3/library/unittest.html#unittest.TestCase.addCleanup) I introduced this TestCase that waits the deletion tasks

Reply:
This is great!